zqd/search/worker.go

```diff
@@ -33,34 +30,24 @@ func NewWorkerOp(req api.WorkerRequest) (*WorkerOp, error) {
 	if req.Dir != -1 {
 		return nil, zqe.E(zqe.Invalid, "time direction must be 1 or -1")
 	}
-
-	chunks := make([]archive.Chunk, len(req.Chunks))
-	for i, chunk := range req.Chunks {
-		id, err := ksuid.Parse(chunk.Id)
-		if err != nil {
-			return nil, zqe.E(zqe.Invalid, "unparsable ksuid")
-		}
-		chunks[i].Id = id
-		chunks[i].First = chunk.First
-		chunks[i].Last = chunk.Last
-		chunks[i].Kind = archive.FileKind(chunk.Kind)
-		chunks[i].RecordCount = chunk.RecordCount
+	store, ok := st.(*archivestore.Storage)
+	if !ok {
+		return nil, zqe.ErrInvalid("unhandled storage type for WorkerOp: %T", store)
 	}
-
-	si := archive.SpanInfo{
-		Span:   req.Span,
-		Chunks: chunks,
+	src, err := store.MultiSource().SourceFromRequest(&req)
```

Comment:
Why is SourceFromRequest a method of driver.MultiSource instead of archivestore.Storage?

Reply:
fyi, @nwt & I discussed some of the different packaging & layout options, and both want some time to think about how this should be structured. @nwt did make an good point that for making requests to the workers, we could use the api client instead of constructing the http requests.